File: 3 Longest Substring/LongestSubstring.java
public class LongestSubstring {
    public int lengthofLongestSubstring(String s){
    int maxLength = 0;
    for (int left = 0, right = 0; right<s.length(); right++){
        int foundIndex = s.indexOf(s.charAt(right), left);
        if(foundIndex != right){
            left = foundIndex + 1;
        }
        maxLength = Math.max(maxLength, right - left + 1);
    }
    return maxLength;
}
}

File: 3 Longest Substring/LongestSubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        lastIndexes = {}
        start, maxLength = 0, 0

        for i, char in enumerate(s):
            if char in lastIndexes:
                lastIndex = lastIndexes[char] + 1
                if lastIndex > start:
                    maxLength = max(maxLength, i - start)
                    start = lastIndex

            lastIndexes[char] = i

        return max(maxLength, len(s) - start)


    